src/ape_ethereum/ecosystem.py

```diff
@@ -252,22 +252,31 @@ def decode_receipt(self, data: dict) -> ReceiptAPI:
         if isinstance(input_data, str):
             input_data = bytes(HexBytes(input_data))
 
+        transaction = BaseTransaction(
```

Comment:
Should not be `BaseTransaction`, needs a method to parse into a real transaction type via `type`

Reply:
[096e8d7](https://github.com/ApeWorX/ape/pull/1015/commits/096e8d76a412bc6d8e8e115de676c3c09cf3b295)